### PR TITLE
Image Augmentation

### PR DESCRIPTION
**Pull request by KubiDemir**
Adding a key **"transform"** in **default_config.yaml** for adding any **torchvision.transforms** image transformations: 
* the types of transforms are specified in order by the list of keys "transform" in default_config.yaml
* these are given in **strings containing the python command** for the respective transformation
* more can be added by listing them as a string after a dash
* If the command takes an **input, it must be included** in the string
* all transformations are **composed automatically** in main.py

### CODE REVIEW

File: default_config.yaml [hunk 1]
```diff
@@ -98,3 +98,5 @@ checkpoint_file_path: "logs/checkpoints"
 # The size to which images should be padded/cropped using PIL.ImageOps.pad
 final_image_size: 500
 
+# The kind of transformation applied: valid options are "default", "random_rotations", "random_transforms"
```
Comment (tmachnitzki) [inline default_config.yaml:101]: The name "default" does not tell at all whats going on. Further how do we combine these transformations here?

File: main.py [hunk 2]
```diff
@@ -42,11 +41,7 @@ def main():
 
     logging.warning(CONFIG.__dict__)  # prints the whole config used for that run
 
-    transform = transforms.Compose([
-        transforms.Pad(CONFIG.final_image_size),
-        transforms.CenterCrop([CONFIG.final_image_size, CONFIG.final_image_size]),
-        transforms.ToTensor(),
-    ])
+    transform = transform_function(CONFIG.transform, CONFIG.final_image_size)
```
Comment (tmachnitzki) [inline main.py:44]: maybe you can rename this to something like "get_tranforms_from_config" or so?

File: src/utils/transformations.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    if transform_type not in all_transform_types:
+        transform_type = "default"
+
+    if transform_type == "default":
+        transform = transforms.Compose(
+            [transforms.Pad(final_image_size), transforms.CenterCrop([final_image_size, final_image_size]),
+             transforms.ToTensor()])
+    elif transform_type == "random_rotations":
+        transform = transforms.Compose(
+            [transforms.RandomRotation(degrees=[-10., 10.]), transforms.Pad(final_image_size),
```
Comment (tmachnitzki) [inline src/utils/transformations.py:16]: Change to 360 degrees

File: train_on_strand.sh [hunk 4]
```diff
@@ -11,7 +11,7 @@
 #SBATCH --cpus-per-task=20
```
Comment (tmachnitzki) [inline train_on_strand.sh:11]: Please do not commit this file

File: train_on_strand.sh [hunk 5]
```diff
@@ -11,7 +11,7 @@
...
```
Comment (tmachnitzki) [inline train_on_strand.sh:12]: Please untrack this file

File: main.py [hunk 6]
```diff
@@ -42,11 +41,7 @@ def main():
 
     logging.warning(CONFIG.__dict__)  # prints the whole config used for that run
 
-    transform = transforms.Compose([
-        transforms.Pad(CONFIG.final_image_size),
-        transforms.CenterCrop([CONFIG.final_image_size, CONFIG.final_image_size]),
-        transforms.ToTensor(),
-    ])
+    transform = transforms.Compose([eval(i) for i in CONFIG.transform])
```
Comment (tmachnitzki) [inline main.py:44]: Awesome!

**Review by tmachnitzki — COMMENTED** [review]
This is just like I imagined it, well done.
Just untrack the slurm file and this is good to go.

**Comment by tmachnitzki** [comment]
To keep the code clean, maybe instead of just merging we can squash and merge, so that this pr just appears as a single commit.

File: train_on_strand.sh (hunk 5 above)
Comment (KubiDemir) [reply train_on_strand.sh]: I did not manage to untrack the file, so  I just copied the train_on_strand from the main branch into mine. Now, there should not be any conflict anymore.

**Pull request merged**